src/python_x509_pkcs11/pkcs11_handle.py

```diff
@@ -118,17 +122,28 @@ def __init__(
             ):
                 self.support_recreate_session = True
 
-        try:
+        # From argument
+        if pkcs11_recreate_session:
+            self.support_recreate_session = True
 
-            self._lib = lib(os.environ["PKCS11_MODULE"])
+        try:
+            # First let us get all 3 PKCS11 variables
+            if not pkcs11_mdoule:
+                pkcs11_mdoule = os.environ["PKCS11_MODULE"]
+            if not pkcs11_token:
+                pkcs11_token = os.environ["PKCS11_TOKEN"]
+            if not pkcs11_pin:
+                pkcs11_pin = os.environ["PKCS11_PIN"]
+
+            self._lib = lib(pkcs11_mdoule)
             self._lib.reinitialize()
 
             # Open the PKCS11 session
-            self._token = self._lib.get_token(token_label=os.environ["PKCS11_TOKEN"])
+            self._token = self._lib.get_token(token_label=pkcs11_token)
```

Comment:
This always confuses me when working with pkcs11, token is short for token-label right?

Reply:
> This always confuses me when working with pkcs11, token is short for token-label right?

Yes.